Devscord.DiscordFramework/Middlewares/Contexts/UserContext.cs

```diff
@@ -1,26 +1,26 @@
-﻿using Devscord.DiscordFramework.Framework.Architecture.Middlewares;
-using System;
+﻿using Devscord.DiscordFramework.Commons;
+using Devscord.DiscordFramework.Framework.Architecture.Middlewares;
 using System.Collections.Generic;
 using System.Linq;
-using System.Text;
 
 namespace Devscord.DiscordFramework.Middlewares.Contexts
 {
     public class UserContext : IDiscordContext
     {
         public ulong Id { get; private set; }
         public string Name { get; private set; }
-        public bool IsAdmin { get; private set; }
         public IEnumerable<UserRole> Roles { get; private set; }
         public string AvatarUrl { get; private set; }
         public string Mention { get; private set; }
 
+        public bool IsAdmin => Roles.Any(x => x.Permissions.Any(x => x.HasFlag(Permission.ManageGuild)));
```

Comment:
to może być ustawiane raz podczas tworzenia instancji 
a nie przy każdym odwołaniu 
bo Roles nigdy sie nie zmienia, to jest obiekt immutable 

co prawda troche stracimy na wydajności, ale operacje na tak małych kolekcjach to nie jest problem